src/bin/test_mpc.rs

```diff
@@ -123,57 +153,80 @@ async fn main() -> Result<(), Box<dyn Error>> {
     }
 
     match args.action {
-        TestAction::Multiply => match args.input.input_type {
-            InputType::Fp31 => {
-                let query_config = QueryConfig {
-                    field_type: FieldType::Fp31,
-                    query_type: QueryType::TestMultiply,
-                };
-                let query_id = clients[0].create_query(query_config).await.unwrap();
-                let output = secure_mul::<Fp31>(input, &clients, query_id).await;
-                print_output(&output);
-            }
-            InputType::Fp32BitPrime => {
-                unimplemented!()
-            }
-            InputType::Int64 => panic!("Only field values are supported"),
-        },
-        TestAction::SemiHonestIPA => {
-            let query_type = QueryType::Ipa(IpaQueryConfig {
-                per_user_credit_cap: 3,
-                max_breakdown_key: 3,
-                num_multi_bits: 3,
-                attribution_window_seconds: None,
-            });
-
-            match args.input.input_type {
-                InputType::Fp31 => {
-                    let query_config = QueryConfig {
-                        field_type: FieldType::Fp31,
-                        query_type,
-                    };
-                    let query_id = clients[0].create_query(query_config).await.unwrap();
-                    let output =
-                        semi_honest::<Fp31, MatchKey, BreakdownKey>(input, &clients, query_id)
-                            .await;
-                    print_output(&output);
-                }
-                InputType::Fp32BitPrime => {
-                    let query_config = QueryConfig {
-                        field_type: FieldType::Fp32BitPrime,
-                        query_type,
-                    };
-                    let query_id = clients[0].create_query(query_config).await.unwrap();
-                    let output = semi_honest::<Fp32BitPrime, MatchKey, BreakdownKey>(
-                        input, &clients, query_id,
-                    )
-                    .await;
-                    print_output(&output);
-                }
-                InputType::Int64 => panic!("Only field values are supported"),
-            }
-        }
+        TestAction::Multiply => multiply(args, &clients).await,
+        TestAction::SemiHonestIPA => semi_honest_ipa(args, &clients).await,
```

Comment:
No Malicious?

Reply:
would be trivial to add, just not there yet